tests/generic_config_updater/test_bgpl.py

```diff
@@ -0,0 +1,170 @@
+import logging
+import pytest
+
+from netaddr import IPNetwork
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.helpers.generators import generate_ip_through_default_route
+from tests.generic_config_updater.gu_utils import apply_patch, expect_op_success
+from tests.generic_config_updater.gu_utils import generate_tmpfile, delete_tmpfile
+from tests.generic_config_updater.gu_utils import create_checkpoint, delete_checkpoint, rollback_or_reload
+
+pytestmark = [
+    pytest.mark.topology('t0'),
+]
+
+logger = logging.getLogger(__name__)
+
+@pytest.fixture(autouse=True)
+def setup_env(duthosts, rand_one_dut_hostname):
+    """
+    Setup/teardown fixture for bgpmon config
+    Args:
+        duthosts: list of DUTs.
+        rand_selected_dut: The fixture returns a randomly selected DuT.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    create_checkpoint(duthost)
+
+    yield
+
+    try:
+        logger.info("Rolled back to original checkpoint")
+        rollback_or_reload(duthost)
+    finally:
+        delete_checkpoint(duthost)
+
+@pytest.fixture(scope="module")
+def bgpmon_setup_info(duthost):
+    """ Get initial setup info for BGPMONITOR
+    """
+    peer_addr = generate_ip_through_default_route(duthost)
+    pytest_assert(peer_addr, "Failed to generate ip address for test")
+    peer_addr = str(IPNetwork(peer_addr).ip)
+
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+    local_addr = mg_facts['minigraph_lo_interfaces'][0]['addr']
+
+    return peer_addr, local_addr, str(mg_facts['minigraph_bgp_asn'])
+
+def bgpmon_cleanup_config(duthost):
+    """ Clean up BGPMONITOR config to make sure t0 is not broken by other tests
+    """
+    cmds = 'sonic-db-cli CONFIG_DB keys "BGP_MONITORS|*" | xargs -r sonic-db-cli CONFIG_DB del'
+    output = duthost.shell(cmds)
+    pytest_assert(not output['rc'],
+        "bgpmon cleanup config failed"
+    )
+
+def check_bgpmon_with_addr(duthost, addr):
+    """ Check BGP MONITOR config change is taken into effect
+    """
+    cmds = "show ip bgp summary | grep -w {}".format(addr)
```

Comment:
>show ip bgp summary [](http://example.com/codeflow?start=12&length=19)

Maybe easier to use `show ip bgp nei {}`

Reply:
`show ip bgp nei` doesn't work on BGPMonitor.
```
admin@vlab-01:~/bgpl$ show ip bgp sum

IPv4 Unicast Summary:
BGP router identifier 10.1.0.32, local AS number 65100 vrf-id 0
BGP table version 4
RIB entries 3, using 576 bytes of memory
Peers 5, using 109080 KiB of memory
Peer groups 5, using 320 bytes of memory


Neighbhor      V     AS    MsgRcvd    MsgSent    TblVer    InQ    OutQ  Up/Down    State/PfxRcd    NeighborName
-----------  ---  -----  ---------  ---------  --------  -----  ------  ---------  --------------  --------------
10.0.0.57      4  64600          0          0         0      0       0  never      Active          ARISTA01T1
10.0.0.59      4  64600          0          0         0      0       0  never      Active          ARISTA02T1
10.0.0.61      4  64600          0          0         0      0       0  never      Active          ARISTA03T1
10.0.0.63      4  64600          0          0         0      0       0  never      Active          ARISTA04T1
11.0.0.1       4  65100          0          0         0      0       0  never      Active          BGPMonitor

Total number of neighbors 5
admin@vlab-01:~/bgpl$ show ip bgp nei 11.0.0.1
Usage: show ip bgp nei [OPTIONS] [IPADDRESS] [[routes|advertised-
                       routes|received-routes]]
Try "show ip bgp nei -h" for help.

Error:  Bgp neighbor 11.0.0.1 not configured

```